src/Apcu/bootstrap_zend.php

```diff
@@ -0,0 +1,119 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+/**
+ * Bootstrap for Zend Server Data Cache.
+ */
+if (!function_exists('apc_add')) {
+    function apcu_add($key, $var = null, $ttl = 0)
+    {
+        if (is_array($key)) {
+            $errors = array();
+            foreach ($key as $cache_key => $cache_value) {
+                if (!apc_add($cache_key, $cache_value, $ttl)) {
+                    $errors[] = $cache_key;
+                }
+            }
+
+            return $errors;
+        }
+
+        return apc_add($key, $var, $ttl);
+    }
+
+    function apcu_store($key, $var = null, $ttl = 0)
+    {
+        if (is_array($key)) {
+            $errors = array();
+            foreach ($key as $cache_key => $cache_value) {
+                if (!apc_store($cache_key, $cache_value, $ttl)) {
+                    $errors[] = $cache_key;
+                }
+            }
+
+            return $errors;
+        }
+
+        return apc_add($key, $var, $ttl);
+    }
+
+    function apcu_exists($keys)
+    {
+        if (is_array($keys)) {
+            $errors = array();
+            foreach ($keys as $cache_key) {
+                if (!apc_exists($cache_key)) {
+                    $errors[] = $cache_key;
+                }
+            }
+
+            return $errors;
+        }
+
+        return apc_exists($keys);
+    }
+
+    function apcu_fetch($key, &$success = false)
+    {
+        if (is_array($key)) {
+            $cache_values = array();
+            foreach ($key as $cache_key) {
+                $success = false;
+                $item = apc_fetch($cache_key, $success);
+                if ($success) {
+                    $cache_values[$cache_key] = $item;
+                }
+            }
+            if (count($cache_values) > 0) {
+                $success = true;
+            }
+
+            return $cache_values;
+        }
+
+        return apc_fetch($key, $success);
+    }
+
+    function apcu_cache_info($limited = false)
+    {
+        return apc_cache_info('user', $limited);
+    }
+
+    function apcu_cas($key, $old, $new)
+    {
+        return apc_cas($key, $old, $new);
+    }
+
+    function apcu_clear_cache()
+    {
+        return apc_clear_cache('user');
+    }
+
+    function apcu_dec($key, $step = 1, &$success = false)
+    {
+        return apc_dec($key, $step, $success);
+    }
+
+    function apcu_delete($key)
+    {
+        return apc_delete($key);
+    }
+
+    function apcu_inc($key, $step = 1, &$success = false)
+    {
+        return apc_inc($key, $step, $success);
+    }
+
+    function apcu_sma_info($limited = false)
+    {
+        return apc_sma_info($limited);
```

Comment:
We should avoid defining all these functions twice, as they are the same when using Zend Server or no